wasm_utils/src/memory_serialization.rs

```diff
@@ -52,8 +52,11 @@ pub fn serialize<T: Serialize>(
 ) -> Result<SinglePageAllocation, RibosomeErrorCode> {
     let json_bytes = serde_json::to_vec(&internal).unwrap();
     let json_bytes_len = json_bytes.len();
-    assert!(json_bytes_len < <u16>::max_value() as usize);
-
+    if json_bytes_len > <u16>::max_value() as usize
+        || json_bytes_len as u32 + stack.top() as u32 > <u16>::max_value() as u32
```

Comment:
This is the 'potential bug' fix, I think? good :)

Reply:
yes!